pkg/util/oom/oom.go

```diff
@@ -0,0 +1,25 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package oom
+
+// This is a struct instead of an interface to allow injection of process ID listers and
+// applying OOM score in tests.
+type OomAdjuster struct {
+	pidLister                 func(cgroupName string) ([]int, error)
```

Comment:
Discussed IRL - the main issue is injecting applyOomScoreAdj. 

As you pointed out, an alternative design is to make an interface and fake for ioutil. NewOomAdjuster would then take in the ioutil interface. In oom_linux_test.go we would then implement and inject a particular type of fake (because we want to inspect exactly which processes the OOM score was applied to). 

I'm going with my current approach because IMO it's shorter/simpler :) Let me know if you think changing this is important, and I will!